### PR TITLE
Core Components Tree

### PR DESCRIPTION
**Pull request by gr2m**
Here’s my attempt of a simple visualisation of how all the hoodie
modules play together.

Immediately a few things come to my mind:
1. All the descriptions could be the same as the summaries in the
   respective READMEs. Seeing them all at once can help to make
   them more clear.
2. Would be nice to make link to the respective modules
3. I wonder if we should rename the `hapi-couchdb-*-api` repos to just `hapi-couchdb-*`, and the current `hapi-couchdb-*` repos to `hoodie-*-server`. Because that's what they are, servers, that combine the hapi plugins and their client counterparts in isolation. I think that would make much more sense. I know renaming libraries / packages is not a good idea, but now is the time to do it, these are not yet used outside hoodie
4. Naming conventions: maybe we should rename
   - pouchdb-hoodie-store to hoodie-client-store
   - account-client to hoodie-client-account
   
   It might make it less clear that these libraries can be used
   by themselves, but that is not really a problem we have at
   this point. Hoodie is complex enough, everything that can help
   us with even adding a little clarity is worth a lot in my opinion

Also note that hapi-couchdb-store-api & hapi-couchdb-task-api do not yet exist, but I'm on it

### CODE REVIEW

**Comment by gr2m** [comment]
Following the dreamcode :tm: philosophy, this is what I'd love to put there:

```
hoodie
│
├─── hoodie-admin-dashboard
│    Web application for admins to manage settings, users and their data.
│
├─┬─ hoodie-server
│ │  Hoodie backend
│ │
│ ├─── hapi-couchdb-account
│ │    Account backend
│ │
│ ├─── hapi-couchdb-store
│ │    Store backend
│ │
│ └─── hapi-couchdb-task
│      Task queue backend
│
└─┬─ hoodie-client
  │  Hoodie api for the browser. Exposed hoodie
  │
  ├─── hoodie-client-log
  │    Log API for the browser. Exposed as hoodie.log
  │
  ├─── hoodie-client-connection-status
  │    Connection API for the browser. Exposed as hoodie.connection
  │
  ├─── hoodie-client-account
  │    Account API for the browser. Exposed as hoodie.account
  │
  ├─── hoodie-client-store
  │    Store API for the browser. Exposed as hoodie.store
  │
  └─── hoodie-client-task
       Task queue API for the browser. Exposed as hoodie.task
```

**Comment by garrensmith** [comment]
This is really nice. And explains it all clearly. I don't quite understand why you need a `hapi-couch-*` and a `hapi-couch-*-api`. From the perspective of someone new to working on hoodie this is really confusing.

**Comment by NickColley** [comment]
Extending some discussion from slack, I'd have thought having hoodie-server-\* might make more sense coupling the client and server together semantically.

At the moment I understand the decision to name them since they're hapi plugins but not sure it's consistent with the client-side.

Not sure of course but it certainly is clearer for me.

```
hoodie
│
├─── hoodie-admin-dashboard
│    Web application for admins to manage settings, users and their data.
│
├─┬─ hoodie-server
│ │  Hoodie backend
│ │
│ ├─── hoodie-server-account
│ │    Account backend
│ │
│ ├─── hoodie-server-store
│ │    Store backend
│ │
│ └─── hoodie-server-task
│      Task queue backend
│
└─┬─ hoodie-client
  │  Hoodie api for the browser. Exposed hoodie
  │
  ├─── hoodie-client-log
  │    Log API for the browser. Exposed as hoodie.log
  │
  ├─── hoodie-client-connection-status
  │    Connection API for the browser. Exposed as hoodie.connection
  │
  ├─── hoodie-client-account
  │    Account API for the browser. Exposed as hoodie.account
  │
  ├─── hoodie-client-store
  │    Store API for the browser. Exposed as hoodie.store
  │
  └─── hoodie-client-task
       Task queue API for the browser. Exposed as hoodie.task
```

**Comment by garrensmith** [comment]
@nickcolley I like that idea. Its definitely more consistent and I would imagine that hoodie server parts being hapi plugins are a nice side effect rather than the main attraction to them.

**Comment by gr2m** [comment]
Okay, I agree with the `hoodie-server-` prefix for the hoodie server core parts. For the Task & Store module, there is no outside usage at this point. But the Account module will likely be used in https://github.com/ehealthafrica/kazana.

But I thought that we could make both, `hoodie-server-*` repos as well as `hapi-couchdb-*`. Even if they would just be a tiny wrapper around a hapi plugin, I think it would be worth for the documentation alone. In the hapi plugin repos, we could document its functionality as an independent hapi plugin, in the `hood-server-*` repos, we could write about how this repo works in the context of hoodie in general. 

tl;dr, I think having the `hoodie-server-` repos would be worth for the sake of better documentation alone.

Any thoughts on this?

**Comment by gr2m** [comment]
As this discussion went from "make a components tree" to "we should make our module naming more coherent", here's a follow up issue with a suggestion and a todo list: https://github.com/hoodiehq/hoodie/issues/436

I'd love to hear your thoughts on it

**Comment by gr2m** [comment]
I've updated the README based on the discussion and the work at #436:
https://github.com/hoodiehq/hoodie/blob/core-components-tree/README.md

How does that look?

**Comment by gr2m** [comment]
Merged this because of today’s "meet the Hoodies" event (https://ti.to/hoodie/hello-nyc/). But please create follow up issues if you have any farther suggestions :)

File: README.md [hunk 1]
```diff
@@ -45,23 +45,29 @@ That's it! Running `npm start` will now serve a hoodie-app from your `www` folde
...
+   4. [client-log](https://github.com/hoodiehq/hoodie-client-log)  
+   5. [client-connection-status](https://github.com/hoodiehq/hoodie-client-connection-status)  
 
-### Core Components
 
-- [**server**](https://github.com/hoodiehq/hoodie-server)
-- [**client**](https://github.com/hoodiehq/hoodie-client)
-- [**admin-dashboard**](https://github.com/hoodiehq/hoodie-admin-dashboard)
+2. [**server**](https://github.com/hoodiehq/hoodie-server)  
+   Hoodie’s magical back-end. It integrates the following hapi plugins:
```
Comment (boennemann) [inline README.md:62]: Sorry for the belated review, but I would appreciate if we could avoid calling things _magical_. The hoodie backend should aim to be simple and well-structured, so everyone can contribute, not just wizards and witches.

Comment (gr2m) [inline README.md:62]: Very good point. I'll fix it

Comment (gr2m) [inline README.md:62]: done via b6695bf

Comment (boennemann) [inline README.md:62]: :+1:
